py/examples/table_commands.py

```diff
@@ -0,0 +1,69 @@
+# Table / Commands
```

Comment:
This example has a few problems:

* Uses global variable `rows` - not a problem when used as read-only, but since it's mutated it results displaying the same amount of rows for each browser (try deleting a row and open a new browser tab, the row count is going to be same instead of initial 50)
* Nit: Unnecessary string casts.
* Unnecessary async functions - whole async/await is relatively complex for regular Wave devs so make sure to use it only when really necessary.
* Deleting a specific row recreates every other row as well so it's impossible to tell whether the delete actually worked compared to previous data. It would be better if only a single row was removed and the rest stayed the same.

Take this as a base (may have some problems that need fixing, but should be generally fine):

```py
# Table / Commands
# Allow group of commands with context menu for each row.
# #table #commands
# ---
from h2o_wave import main, app, Q, ui
from faker import Faker

fake = Faker()


class TableRow:
    _id = 0

    def __init__(self):
        TableRow._id += 1
        self.id = f'row_{TableRow._id}'
        self.name = f'{fake.first_name()} {fake.last_name()}'
        self.details = fake.sentence()


def show_table(q) -> None:
    q.page['example'] = ui.form_card(box='1 1 4 -1', items=[
        ui.table(
            name='table',
            columns=[
                ui.table_column(name='name', label='Name'),
                ui.table_column(
                    name='actions', label='Actions',
                    cell_type=ui.command_table_cell_type(name='commands', items=[
                        ui.command(name='details', label='Details'),
                        ui.command(name='delete', label='Delete'),
                    ])
                )
            ],
            rows=[ui.table_row(name=r.id, cells=[r.name]) for r in q.client.rows]
        )
    ])


@app('/demo')
async def serve(q: Q):
    if not q.app.initialized:
        q.app.rows = [TableRow() for _ in range(50)]
        q.app.initialized = True
    if not q.client.initialized:
        q.client.rows = q.app.rows
        show_table(q)
        q.client.initialized = True

    if q.args.delete:
        q.client.rows = [row for row in q.client.rows if row.id != q.args.delete]
        q.page['example'].items[0].table.rows = [ui.table_row(name=r.id, cells=[r.name]) for r in q.client.rows]
    if q.args.details:
        for row in q.client.rows:
            if row.id == q.args.details:
                q.page['example'] = ui.form_card(box='1 1 4 4', items=[
                    ui.text(name='details', content=row.details),
                    ui.button(name='back', label='Back')
                ])
                break
    if q.args.back:
        show_table(q)

    await q.page.save()
```

Reply:
> Unnecessary async functions - whole async/await is relatively complex for regular Wave devs so make sure to use it only when really necessary.

I based it on `todo.py` but I will keep that in mind.

Reply:
I couldn't spot any problem with your example, works just fine. Thanks.